include/fixed_containers/fixed_red_black_tree_view.hpp

```diff
@@ -0,0 +1,386 @@
+}
+
+class FixedRedBlackTreeRawView
+{
+    using Compactness = fixed_red_black_tree_detail::RedBlackTreeNodeColorCompactness;
+    using CompactnessBacking = fixed_red_black_tree_detail::detail::RedBlackTreeNodeColorCompactnessBackingEnum;
+    using StorageType = fixed_red_black_tree_detail::RedBlackTreeStorageType;
+    using NodeIndex = fixed_red_black_tree_detail::NodeIndex;
+
+    static constexpr auto NULL_INDEX = fixed_red_black_tree_detail::NULL_INDEX;
+    
+public:
+    class Iterator
+    {
+    public:
+        using value_type = const std::byte*;
+        using difference_type = std::ptrdiff_t;
+        using pointer = value_type*;
+        using const_pointer = const value_type*;
+        using reference = value_type&;
+        using const_reference = const value_type&;
+        using iterator_category = std::forward_iterator_tag;
+
+        constexpr Iterator(const std::byte* ptr,
+                           std::size_t elem_size_bytes,
+                           std::size_t max_size_bytes,
+                           Compactness compactness,
+                           StorageType storage_type,
+                           bool end = false) noexcept
+          : base_(ptr)
+          , elem_size_bytes_(elem_size_bytes)
+          , max_size_bytes_(max_size_bytes)
+          , compactness_(compactness)
+          , storage_type_(storage_type)
+          , storage_elem_size_bytes_(storage_elem_size_bytes())
+          , index_(end ? NULL_INDEX : min_index())
+          , cur_pointer_(node_pointer(index_))
+        {}
+
+        constexpr Iterator() noexcept
+          : Iterator(nullptr, {}, {}, {}, {}, false)
+        {}
+
+        constexpr Iterator(const Iterator&) noexcept = default;
+        constexpr Iterator(Iterator&&) noexcept = default;
+        constexpr Iterator& operator=(const Iterator &) noexcept = default;
+        constexpr Iterator& operator=(Iterator&&) noexcept = default;
+
+        constexpr const_reference operator*() const
+        {
+            return cur_pointer_;
+        }
+        
+        constexpr const_pointer operator->()
+        {
+            return &cur_pointer_;
+        }
+        
+        constexpr Iterator& operator++()
+        {
+            index_ = successor(index_);
+            cur_pointer_ = node_pointer(index_);
+            return *this;
+        }
+
+        constexpr Iterator operator++(int) & noexcept
+        {
+            auto tmp = *this;
+            index_ = successor(index_);
+            cur_pointer_ = node_pointer(index_);
+            return tmp;
+        }
+        
+        constexpr bool operator==(const Iterator& other) const
+        {
+            return (
+                base_ == other.base_ &&
+                index_ == other.index_
+            );
+        }
+        
+    private:
+        /**
+         * Calculate the pointer to a tree node at the provided storage index.
+         */
+        constexpr const std::byte* node_pointer(std::size_t i) const
```

Comment:
Nit: can be marked `[[nodiscard]]`.
Applies to all private functions below.

Reply:
This function is not transferring any ownership so I don't think we need that here. I will add that annotation to catch any unintended use of these private functions, though.

Reply:
83b640c